server/imports/lab/methods.ts

```diff
@@ -12,7 +12,11 @@ var LabSession = require('../api/lab.session.js');
 import{ prepLab, next, verify } from './labMethods.ts';
 
 Meteor.methods({
-  /**prepareLab: prepares a labExec object for the current user
+ 
+   'uploadLab': function(lab){
+     Collections.labs.insert(lab);
```

Comment:
1) This function lets literally any user upload a lab with authenticated permissions, which is quite evil.
2) You don't need this.  The client should have access to a Collections.labs.insert method which does the same thing already